### PR TITLE
Perf: minor optimization in module_lr

### PR DESCRIPTION
**Pull request by maki49**
- Do not produce "running_scf.log" when esolver_type==lr
- Fix a memory waste in `LR_Util::matsym`

### CODE REVIEW

File: source/module_base/global_file.cpp [hunk 1]
```diff
@@ -153,28 +153,18 @@ void ModuleBase::Global_File::make_dir_out(
...
+        {
+            std::stringstream ss;
+            ss << "running_" << keyword;
+            if (out_alllog) { ss << "_" << rank + 1; }
+            open_log(GlobalV::ofs_running, ss.str(), keyword, restart);
+#if defined(__CUDA) || defined(__ROCM)
+            open_log(GlobalV::ofs_device, "device" + std::to_string(rank), calculation, restart);
+#endif
+        };
+    // if linear response solver only, open running_lr.log. else, open running_calculation.log
```
Comment (mohanchen) [inline source/module_base/global_file.cpp]: what is the exact meaning of 'calculation?' why we need two names for lr? you write the code in a new way that makes lr so special, it's strange

Comment (mohanchen) [reply source/module_base/global_file.cpp]: it's similar to what Peize wrote for the old exx code:   if(exx) {xxx} else {yyy}, here yyy is what most users are familiar with, such as scf and nscf, and 'xxx' is only a special case in my point of view. It's not good to write code in this special way to emphasize the exx code.

**Comment by mohanchen** [comment]
‘lr' should not be a special case that appears in deciding the name of running log. This is not a good enough design. I would suggest the developer to think a more appropriate way to implement the function.

**Comment by maki49** [comment]
> ‘lr' should not be a special case that appears in deciding the name of running log. This is not a good enough design. I would suggest the developer to think a more appropriate way to implement the function.

I've changed the code to the new solution to avoid log-coverage: if `esolver_type=="lr"`, modify `calculation` to `"nscf"`.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**